File: MsSqlSemaphore/DocNetC8/PvWay.PgSqlSemaphoreService.nc8/Services/ISemaphoreServiceConfig.cs
namespace PvWay.PgSqlSemaphoreService.nc8.Services;

internal interface ISemaphoreServiceConfig
{
    string SchemaName { get; }
    string TableName { get; }
    Func<Task<string>> GetCsAsync { get; }
    Action<Exception> LogException { get; }
    Action<string>? LogInfo { get; } 
}

File: MsSqlSemaphore/DocNetC8/PvWay.PgSqlSemaphoreService.nc8/Services/ISemaphoreServiceConfig.py
from typing import Callable, Optional
from abc import ABC, abstractproperty

class ISemaphoreServiceConfig(ABC):
    @property
    @abstractproperty
    def SchemaName(self) -> str:
        pass

    @property
    @abstractproperty
    def TableName(self) -> str:
        pass

    @property
    @abstractproperty
    def GetCsAsync(self) -> Callable[[], 'Task[str]']:
        pass

    @property
    @abstractproperty
    def LogException(self) -> Callable[[Exception], None]:
        pass

    @property
    @abstractproperty
    def LogInfo(self) -> Optional[Callable[[str], None]]:
        pass